tools/scripts/script.ps1

```diff
@@ -0,0 +1,242 @@
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param(
+    [Parameter(HelpMessage = "Specify this parameter to uninstall AIShell")]
+    [switch] $Uninstall
+)
+
+<#################################################
+#
+#               Helper functions
+#
+#################################################>
+
+function Get-PowerShellVersion {
+    if ([version]$PSVersionTable.PSVersion -lt [version]"7.4.6") {
+        Write-Warning "[PowerShell version 7.4.6 is the minimum requirement for AIShell module, exiting...]"
+        return $False
+    }
+    return $True
+}
+
+function Get-InstallDirectoryForCurrentOS {
+    $destinationDirectoryForMac = Join-Path "usr" "local" "AIShell"
+    $destinationDirectoryForWindows = Join-Path $env:LOCALAPPDATA "Programs" "AIShell"
+    # check os
+    if ($IsWindows) {
+        return $destinationDirectoryForWindows
+    } 
+    elseif ($IsMacOS) {
+        return $destinationDirectoryForMac
+    }
+    else {
+        Write-Warning "This install script is only compatible with Windows and Mac systems, if you want to install on Linux please download the package directly from the GitHub repo at aka.ms/AIShell-Repo"
+        return ""
+    }
+}
+
+function Save-AIShellFromUrl {
+    # using placeholder urls until the GitHub release files are ready
+    $osxArm64Url = "osxArm64Url"
+    $osxX64Url = "osxX64Url"
+    $winArm64Url = "winArm64Url"
+    $winX64Url = "winX64Url"
+    $winX86Url = "winX86Url"
+    $moduleUrl = "moduleUrl"
+
+    if (!$moduleUrl) {
+        return
+    }
+
+    # Determine download url based on OS and architecture
+    $destinationDirectory = Get-InstallDirectoryForCurrentOS
+    $osArchitecture = [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture
+
+    $buildUrl = ""
+    if ($IsWindows) {
+        if ($osArchitecture -eq "X64") {
+            $buildUrl = $winX64Url
+        } 
+        elseif ($osArchitecture -eq "X86") {
+            $buildUrl = $winX86Url
+        } 
+        elseif ($osArchitecture -eq "Arm64") {
+            $buildUrl = $winArm64Url
+        } 
+    } 
+    elseif ($IsMacOS) {
+        if ($osArchitecture -eq "X64") {
+            $buildUrl = $osxX64Url
+        } 
+        elseif ($osArchitecture -eq "Arm64") {
+            $buildUrl = $osxArm64Url
+        } 
+    }
+
+    # Create the directory if not existed
+    if(!(Test-Path -path $destinationDirectory)) {
+        New-Item -Path $destinationDirectory -ItemType Directory -Force | Out-Null
+    }
+
+    # Download AIShell Module
+    $tempFilesDirectory = Join-Path $destinationDirectory "temp"
```

Comment:
module will be installed from PowerShellGallery with `Install-Module`.

Reply:
![image](https://github.com/user-attachments/assets/8d0a55c9-aa56-44d3-bb8e-161ae9409ca9)

Did you actually check the packages I shared with you? The macOS package is a `.tar.gz` file, and `Expand-Archive` doesn't work for it.